test/unit/test_util.c

```diff
@@ -62,7 +62,28 @@ bool test_leading_zeros(void) {
 	mu_end;
 }
 
+
+bool test_htpp_anomaly(void) {
+	HtPP *ht = ht_pp_new0();
+	void *p1 = malloc(0x10);
+	void *p2 = malloc(0x20);
+	void *p3 = malloc(0x30);
+	ut64 p4 = 0x50505050505;
+	ht_pp_insert(ht, p1, (void *) 1);
```

Comment:
this doesn't work because it will threat `p1`, `p2`, `p3` and `p4` as strings instead of raw pointers.
You need to implement a custom hash function in order to calculate the hash.

Reply:
to use raw pointers as actual input you need to use `HtUP` and cast `void*` to `ut64`

Reply:
Yeah, I got to it from `SetP`. Which uses it (incorrectly) like this. I would say to get rid of the `ht_xx_new0()` functions. First of all, they are not documented. And figuring out how the hash table works exactly, is pretty hard because of all the macros used.
So it is in-transparent, if it works or not, hence one would just assume everything is fine.

Reply:
i actually suggest to make proper headers and implement correctly the methods and provide a ht_ss/ht_su/ht_sp types.

Reply:
Or predefined set of options